### ISSUE TITLE
Extra , is added after tail in head tail list pattern

### ISSUE THREAD

**Issue by KronicDeth** [uohzxela/ex_format#90]
# Current Bahavior

![screen shot 2017-09-01 at 8 57 54 am](https://user-images.githubusercontent.com/298259/29973049-cd176b8a-8ef3-11e7-8db7-b4f5d15090c4.png)

## IntelliJ Elixir

The multiple heads in the `[head | tail]` match pushed it over the column limit, so every parameter wraps.

```elixir
  def can?(
          %InterpreterProfile{id: id},
          :show,
          [
            %ActualsNote{job_assignment_id: job_assignment_id},
            %JobAssignment{id: job_assignment_id, interpreter_profile_id: id}
            |
            _
          ]
        ), do: true
```

## `ex_format`

Pro: it converts the `do: true` to `do true end` because other clauses have blocks.
Con: It adds a `,` after the `_` in the head tail match.

A trailing comma is allowed in a list, but it looks weird to me because it's after the `_` that is matching the tail, so it reads to me there is the tail `_` and then something after it after the `,` that was inserted.

```elixir
def can?(%InterpreterProfile{id: id}, :show, [
    %ActualsNote{job_assignment_id: job_assignment_id},
    %JobAssignment{id: job_assignment_id, interpreter_profile_id: id} | _,
  ]) do
    true
  end
```

**Comment by lpil** [comment]
I'm unsure if this is the correct behaviour as per the style guide. Perhaps we should open an issue there

**Comment by uohzxela** [comment]
@KronicDeth thanks for the bug report! Please have a look at the latest incarnation of ExFormat [here](https://github.com/whatyouhide/code_formatter). It's written by @josevalim and @whatyouhide themselves, and is going to be included in Elixir master for the v1.6 release (methinks). Their formatter handles the edge cases more effectively as they are using a more generalized algorithm for line wrapping. They also made a lot of changes to Elixir source with regards to the tokenizer and AST parser to preserve more information from the source code.

Note: you need to run on Elixir master to use their formatter.

Here's the current behavior of their formatter (it [doesn't support trailing comma anymore](https://github.com/lexmag/elixir-style-guide/pull/46#pullrequestreview-60077938)):

```
def can?(%InterpreterProfile{id: id}, :show, [
      %ActualsNote{job_assignment_id: job_assignment_id},
      %JobAssignment{id: job_assignment_id, interpreter_profile_id: id} | _
    ]), do: true
```

It's encouraged that you submit future issues directly to the linked repository since José is actively working on it nowadays and he can answer your queries.
